homeassistant/components/modbus/validators.py

```diff
@@ -266,15 +288,28 @@ def duplicate_entity_validator(config: dict) -> dict:
                 addr += "_" + str(inx)
                 entry_addrs: set[str] = set()
                 entry_addrs.add(addr)
-
-                if CONF_TARGET_TEMP in entry:
+                if CONF_TARGET_TEMP in entry and isinstance(
```

Comment:
The duplicate validator run last, so the value if present is a list.

Reply:
@janiversen When I tested this piece of code on a real env, I obtained an error caused by the iteration over an int, so it seems to me that it run before the validation. Am I wrong?

Reply:
Well your validator changes the config, like some of the others, so it is important it runs after those....if it does not correct it.

Reply:
@janiversen I spent some time using the debugger and also the other validators which change the output (i.e. number_validator or nan..) is called AFTER the duplicate_entity_validator. 
What I miss?!

Reply:
You miss that you need to change that.

Reply:
Dear @janiversen I really appreciate your patience and your (+/-) hidden suggestions; I tried to change the code every time you asked and I can assure that imagining what you have in mind was not so easy. But honestly speaking, this time I cannot rewrite/verify the whole validation process just to avoid an "if/else" inside a routine (we'll have the same for fan_mode conf address of the other PR). 
The debugger says that it is not true that _"The duplicate validator run last [...]"_ 
Domain level duplicate_entity_validator is executed BEFORE all, standard or custom field validators and in fact it is executed against fake values, it the config is wrong. I don't know if it is right, because IMHO is totally unuseful to check duplicates if the entity itself is wrong, but this is not my matter and I'm sure that who wrote that had a good reason.. 
If you have already an idea on how we should change the whole validation process, please, explain.. Else, I commit as it is this piece of code.

Reply:
Its a simple problem of looking in __init__, not a rewrite.